### ISSUE TITLE
Incorrect line number in test

### ISSUE THREAD

**Issue by minton1** [nomiclabs/hardhat#636]
Repro steps: 

Create this test in a file called "`test/SanityTest.ts`"
```
import { ethers } from "@nomiclabs/buidler";
import chai from "chai";
const { expect } = chai;

describe("Sanity Test", () => {

  it("Should succeed.", async () => {
    expect(5).to.equal(5);
  });

  it("Should fail.", async () => {
    expect(5).to.equal(3);
  });
});
```

run `npx buidler test test/SanityTest.ts`

Expected result: 
Test fails indicating the correct line for the failure, **line 12.**

Actual result: 

**line 53** is indicated:
```
Sanity Test
    ✓ Should succeed.
    1) Should fail.


  1 passing (12ms)
  1 failing

  1) Sanity Test
       Should fail.:

      AssertionError: expected 5 to equal 3
      + expected - actual

      -5
      +3

      at /Users/mm/Documents/GitHub/project/test/SanityTest.ts:53:26
      at step (test/SanityTest.ts:33:23)
      at Object.next (test/SanityTest.ts:14:53)
      at /Users/mm/Documents/GitHub/project/test/SanityTest.ts:8:71
      at new Promise (<anonymous>)
      at __awaiter (test/SanityTest.ts:4:12)
      at Context.<anonymous> (test/SanityTest.ts:51:45)
      at processImmediate (internal/timers.js:443:21)
```

**Comment by minton1** [comment]
setup: 

```
mac os 10.15.4
node -v v11.15.0

  {
    "dependencies": {
      "@nomiclabs/buidler": "^1.3.3",
      "@nomiclabs/buidler-ethers": "^1.3.3",
      "@nomiclabs/buidler-etherscan": "^1.3.3",
      "@nomiclabs/buidler-truffle5": "^1.3.3",
      "@nomiclabs/buidler-waffle": "^1.3.3",
      "@nomiclabs/buidler-web3": "^1.3.3",
      "@openzeppelin/contracts": "^3.0.1",
      "@studydefi/money-legos": "^2.2.1",
      "@types/chai": "^4.2.11",
      "@types/mocha": "^7.0.2",
      "@types/node": "^14.0.1",
      "@uniswap/v2-core": "^1.0.0",
      "@uniswap/v2-periphery": "^1.0.0-beta.0",
      "buidler-typechain": "^0.0.5",
      "chai": "^4.2.0",
      "ethereum-waffle": "^2.3.0",
      "ethers": "^4.0.23",
      "solidity-cli": "^1.0.3",
      "solidity-coverage": "^0.7.5",
      "ts-generator": "^0.0.8",
      "ts-node": "^8.10.1",
      "typechain": "^1.0.3",
      "typechain-target-ethers": "^1.0.4",
      "typechain-target-truffle": "^1.0.2",
      "typechain-target-web3-v1": "^1.0.4",
      "typescript": "^3.9.2",
      "web3": "^1.2.7"
    }
  }
```

**Comment by alcuadrado** [comment]
Can you try the latest version of the waffle plugin?

**Comment by minton1** [comment]
Updated to as follows, issue still repros. Notice these packages, which are now updated to the latest version

`"@nomiclabs/buidler-waffle": "1.3.4",`
`"ethereum-waffle": "2.5.1",`

```
npm manifest:
  {
    "dependencies": {
      "@nomiclabs/buidler": "^1.3.3",
      "@nomiclabs/buidler-ethers": "^1.3.3",
      "@nomiclabs/buidler-etherscan": "^1.3.3",
      "@nomiclabs/buidler-truffle5": "^1.3.3",
      "@nomiclabs/buidler-waffle": "1.3.4",
      "@nomiclabs/buidler-web3": "^1.3.3",
      "@openzeppelin/contracts": "^3.0.1",
      "@studydefi/money-legos": "^2.2.1",
      "@types/chai": "^4.2.11",
      "@types/mocha": "^7.0.2",
      "@types/node": "^14.0.1",
      "@uniswap/v2-core": "^1.0.0",
      "@uniswap/v2-periphery": "^1.0.0-beta.0",
      "buidler-typechain": "^0.0.5",
      "chai": "^4.2.0",
      "ethereum-waffle": "2.5.1",
      "ethers": "^4.0.23",
      "solidity-cli": "^1.0.3",
      "solidity-coverage": "^0.7.5",
      "ts-generator": "^0.0.8",
      "ts-node": "^8.10.1",
      "typechain": "^1.0.3",
      "typechain-target-ethers": "^1.0.4",
      "typechain-target-truffle": "^1.0.2",
      "typechain-target-web3-v1": "^1.0.4",
      "typescript": "^3.9.2",
      "web3": "^1.2.7"
    }
  }
```

**Comment by minton1** [comment]
Issue is solved by removing this line from buidler config: 

usePlugin("solidity-coverage");

**Comment by alcuadrado** [comment]
This has been reported upstream: https://github.com/sc-forks/solidity-coverage/issues/514

**Comment by cgewecke** [comment]
This should be fixed with solidity-coverage `0.7.7`. 

Thanks @alcuadrado for diagnosing & reporting!

**Comment by alcuadrado** [comment]
Thanks, @cgewecke !

Can you try this new version and confirm if it fixed your problem, @minton1 ? Thanks
